ql/src/java/org/apache/hadoop/hive/ql/io/HiveInputFormat.java

```diff
@@ -890,6 +894,24 @@ public static void pushFilters(JobConf jobConf, TableScanOperator tableScan,
     Utilities.setColumnTypeList(jobConf, tableScan);
     // push down filters
     ExprNodeGenericFuncDesc filterExpr = scanDesc.getFilterExpr();
+    String pruningFilter = jobConf.get(TableScanDesc.PARTITION_PRUNING_FILTER);
+    // If we have a pruning filter then combine it with the original
+    if (pruningFilter != null) {
+      ExprNodeGenericFuncDesc pruningExpr = SerializationUtilities.deserializeExpression(pruningFilter);
+      if (filterExpr != null) {
+        // Combine the 2 filters with AND
+        filterExpr = new ExprNodeGenericFuncDesc(TypeInfoFactory.booleanTypeInfo, new GenericUDFOPAnd(), "and",
```

Comment:
note: you could probably use `ExprNodeDescUtils#conjunction` (or move this method there...)

Reply:
Thanks for pointing this out!
Used `and`